src/system/mod.rs

```diff
@@ -267,6 +267,17 @@ fn inject_group(target: GroupId, groups: &mut Vec<GroupId>) {
     }
 }
 
+/// Set the supplementary groups -- returns a c_int to mimic a libc function
+fn set_supplementary_groups(groups: &[GroupId]) -> libc::c_int {
```

Comment:
Maybe push the `cerr` calls into this function?

Reply:
I kind of didn't do that on purpose since it's always used in conjunction with the `setuid/setgid` functions (and this function requires special care anyway); but if you prefer also having io::Result here that's of course a simple change

Reply:
https://github.com/trifectatechfoundation/sudo-rs/compare/a7d54bd594fa3a6ab016b3495b5492d39868989f..fadbfc45e7244cdac6469641d103d4c0235f9c47